### PR TITLE
Add remote functionality to python/oxen

### PR DESCRIPTION
**Pull request by benartuso**

### CODE REVIEW

File: oxen/src/py_remote_repo.rs [hunk 1]
```diff
@@ -95,18 +108,118 @@ impl PyRemoteRepo {
...
+                &self.revision,
+                &user_id,
+                &path,
+                liboxen::constants::DEFAULT_PAGE_NUM,
+                liboxen::constants::DEFAULT_PAGE_SIZE,
+            )
+            .await
+        })?;
+
+        let mut status = StagedData::empty();
```
Comment (gschoeni) [inline oxen/src/py_remote_repo.rs:202]: It feels like we should do this in a From trait on the PyStagedData struct

File: oxen/src/py_remote_repo.rs [hunk 2]
```diff
@@ -123,4 +236,49 @@ impl PyRemoteRepo {
...
+        match branch {
+            Ok(branch) => Ok(PyBranch {
+                name: branch.name,
+                commit_id: branch.commit_id,
+            }),
+            _ => Err(PyValueError::new_err("Could not get or create branch")),
+        }
+    }
+    fn checkout(&mut self, revision: String) -> PyResult<()> {
+        // TODO: Flatten
```
Comment (gschoeni) [inline oxen/src/py_remote_repo.rs:263]: Agreed, feels like we can avoid nesting here

**Review by gschoeni — CHANGES_REQUESTED** [review]
Minor aesthetic changes, but otherwise excited to get this in!

File: oxen/src/py_remote_repo.rs (hunk 2 above)
Comment (benartuso) [reply oxen/src/py_remote_repo.rs]: Updated, lmk if you had something different in mind for the commit case!

File: oxen/src/py_remote_repo.rs (hunk 1 above)
Comment (benartuso) [reply oxen/src/py_remote_repo.rs]: done!

**Review by gschoeni — APPROVED** [review]

File: oxen/src/py_remote_repo.rs (hunk 1 above)
Comment (gschoeni) [reply oxen/src/py_remote_repo.rs]: beautiful

**Pull request merged**